libs/auth/src/card_reader.ts

```diff
@@ -0,0 +1,148 @@
+import { Buffer } from 'buffer';
+import pcscLite from 'pcsclite';
+import { promisify } from 'util';
+import { assert } from '@votingworks/basics';
+import { isByte } from '@votingworks/types';
+
+import {
+  CommandApdu,
+  GET_RESPONSE,
+  MAX_APDU_LENGTH,
+  ResponseApduError,
+  STATUS_WORD,
+} from './apdu';
+
+type BaseTransmit = (data: Buffer) => Promise<Buffer>;
+
+type PcscLite = ReturnType<typeof pcscLite>;
+
+type ReaderStatus = 'error' | 'no_card_reader' | 'no_card' | 'ready';
+
+/**
+ * A class for interfacing with a smart card reader, implemented using PCSC Lite
+ */
+export class CardReader {
+  private baseTransmit?: BaseTransmit;
+  private readonly onReaderStatusChange: (readerStatus: ReaderStatus) => void;
+  private readonly pcscLite: PcscLite;
+  private readerStatus: ReaderStatus;
+
+  constructor(input: {
+    onReaderStatusChange: (readerStatus: ReaderStatus) => void;
+  }) {
+    this.baseTransmit = undefined;
+    this.onReaderStatusChange = input.onReaderStatusChange;
+    this.pcscLite = pcscLite();
+    this.readerStatus = 'no_card_reader';
+
+    this.pcscLite.on('error', () => {
+      this.updateReaderStatus('error');
+    });
+
+    this.pcscLite.on('reader', (reader) => {
+      reader.on('error', () => {
+        this.updateReaderStatus('error');
+      });
+
+      reader.on('status', (status) => {
```

Comment:
In case it wasn't clear, even today with memory cards and `services/smartcards`, though we poll the service (by way of the frontend polling the backend), the service listens for card reader events:

https://github.com/votingworks/vxsuite/blob/e2320afaf33bf81f75ce1f7917c313ac454ace0c/services/smartcards/smartcards/card.py#L329-L330

The same is true in this new code path. The `CardReader` class will listen for card reader events.